test/e2e/es/stackconfigpolicy_test.go

```diff
@@ -67,15 +67,30 @@ func TestStackConfigPolicy(t *testing.T) {
 	assert.NoError(t, err)
 
 	// list of endpoints to check the existence or not of the settings defined in the esConfigSpec
-	configuredObjectsEndpoints := []string{
-		"/_snapshot/repo_test",
-		"/_slm/policy/slm_test",
-		//"/_security/role_mapping/role_test", uncomment once https://github.com/elastic/elasticsearch/issues/91939 is resolved
-		"/_ingest/pipeline/pipeline_test",
-		"/_ilm/policy/ilm_test",
-		"/_index_template/template_test",
-		"/_component_template/runtime_component_template_test",
-		"/_component_template/component_template_test",
+	configuredObjectsEndpoints := map[string]statusCode{
+		"/_snapshot/repo_test":  {ok, notFound},
+		"/_slm/policy/slm_test": {ok, notFound},
+		"/_security/role_mapping/role_test": {
+			onCreation: func() int {
+				// due to a bug in 8.15.x, role mappings are not found via the API
+				if stackVersion.GTE(version.MinFor(8, 15, 0)) && stackVersion.LT(version.MinFor(8, 15, 3)) {
+					return notFound()
+				}
+				return ok()
+			},
+			onDeletion: func() int {
+				// prior to 8.15.x, role mappings are not deleted when scp is deleted
+				if stackVersion.LT(version.MinFor(8, 15, 0)) {
+					return ok()
+				}
+				return notFound()
+			},
+		},
```

Comment:
Thinking back, I think I should move the role mapping out of this map and do a specific validation for it.

Reply:
You could do a custom thing where we inspect cluster state instead. It is not considered a stable interface but might be still be OK to see if the role mappings are there in a test. As we know now that does not guarantee that they are also applied (at least no in the buggy versions)

Reply:
I rewrote a new version in 3149c08 that uses `/_cluster/state/metadata?filter_path=metadata.role_mappings`.